main.go

```diff
@@ -136,64 +137,18 @@ func main() {
 					failf("Failed to write in tabwriter, error: %s", err)
 				}
 
-				for _, step := range responseModel.Steps {
+				sortedSteps := makeSortedCopyOfSteps(responseModel.Steps)
+				doesRetriesForFlakiness := configs.FlakyTestAttempts > 0
+				for index, step := range sortedSteps {
+					isLastStep := index == len(responseModel.Steps)-1
```

Comment:
Is this logic good? I mean the steps could contain multiple different test executions and should not we take the last one of the same test execution.?

If that is the case then maybe a simpler logic should be to check if the flaky retry was enabled like you did. Then if it was we could first deduplicate the array by keeping the last execution of the tests. And then iterate over them.  This way we could even skip the many conditions in the `getNewSuccessValue` function.